src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -547,41 +547,116 @@ private void SetDirty(TimeSpan? maxDelay)
+            var startupExceptions = new List<Exception>();
+
+            int attempts = 0;
+
             try
             {
-                await ExecuteWithFailOverPolicyAsync(
-                    availableClients,
-                    async (client) =>
+                while (true)
+                {
+                    attempts++;
+
+                    IEnumerable<ConfigurationClient> clients = _configClientManager.GetAvailableClients(DateTimeOffset.UtcNow, ignoreBackoff: true);
+
+                    if (await TryInitializeAsync(clients, startupExceptions, cancellationToken).ConfigureAwait(false))
                     {
-                        data = await LoadSelectedKeyValues(
-                            client,
-                            cancellationToken)
-                            .ConfigureAwait(false);
-
-                        watchedSettings = await LoadKeyValuesRegisteredForRefresh(
-                            client,
-                            data,
-                            cancellationToken)
-                            .ConfigureAwait(false);
-
-                        watchedSettings = UpdateWatchedKeyValueCollections(watchedSettings, data);
-                    },
-                    cancellationToken)
-                    .ConfigureAwait(false);
+                        break;
+                    }
+
+                    try
+                    {
+                        await Task.Delay(FailOverConstants.MinStartupBackoffDuration.CalculateBackoffDuration(FailOverConstants.MaxBackoffDuration, attempts), cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        throw new TimeoutException($"The provider timed out while attempting to load data on startup.",
+                            new AggregateException(startupExceptions));
+                    }
+                }
             }
             catch (Exception exception) when (
                 ignoreFailures &&
                 (exception is RequestFailedException ||
+                exception is KeyVaultReferenceException ||
+                exception is TimeoutException ||
                 exception is OperationCanceledException ||
                 ((exception as AggregateException)?.InnerExceptions?.Any(e =>
                     e is RequestFailedException ||
                     e is OperationCanceledException) ?? false)))
             { }
+        }
+
+        private async Task<bool> TryInitializeAsync(IEnumerable<ConfigurationClient> clients, List<Exception> startupExceptions, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                await InitializeAsync(clients, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception) when (exception is OperationCanceledException ||
+                ((exception as AggregateException)?.InnerExceptions?.Any(e =>
```

Comment:
aggregate exception is handled below. Seems this should just be catch operationcanceledexception.

Reply:
We've had customer issues where the SDK returns an `AggregateException` that contains only `OperationCanceledException`, so I wanted to check for that in case it was thrown because of our startup cancellation token. I thought it might avoid throwing an `AggregateException` when it was actually cancelled by the timeout and should be a `TimeoutException`.

Reply:
That's fine, but should be handled in the catch block for aggregate exception.